### PR TITLE
Stop breaking the back button after a reload

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

File: test/features/10-functional/10-basic/50-history.feature [hunk 1]
```diff
@@ -3,10 +3,9 @@ Feature: History navigation
     # As a user
     # When I visit a page
     # I want to be able to use the back button to navigate
-
+    @newSession
```
Comment (danni) [inline test/features/10-functional/10-basic/50-history.feature]: This is not super BDD.

File: src/client.js [hunk 2]
```diff
@@ -25,12 +25,22 @@ function History() {
...
     function goBack() {
-        if (historyLength > 0) {
-            historyLength--;
-            history.goBack();
-        } else {
-            history.pushState(null, "/");
-        }
+        const currentUrl = location.toString();
+
+        historyLength--;
```
Comment (danni) [inline src/client.js]: What happens if the history length goes negative? Why even keep the length if we don't check it?

File: test/features/10-functional/10-basic/50-history.feature (hunk 1 above)
Comment (DanielHeath) [inline test/features/10-functional/10-basic/50-history.feature]: Would 'Given I am using a new browser' be better?

Comment (danni) [inline test/features/10-functional/10-basic/50-history.feature]: Yes.

File: test/steps/browser.js [hunk 3]
```diff
@@ -247,11 +249,37 @@ async function assertItemNotChecked(label: string): Promise<void> {
     assert.equal(checked, null);
 }
 
+async function newBrowser(): Promise<void> {
```
Comment (danni) [inline test/steps/browser.js:258]: Needs a jsdocstring.

**Comment by danni** [comment]
What happens now with deeplinking?

**Comment by danni** [comment]
I.e. This link here, where should Back go? https://askizzy-staging.infoxchangeapps.net.au/service/1078023-melbourne-youth-support-service-myss-at-frontyard-youth-services

**Comment by DanielHeath** [comment]
@danni I think this is now the best we can do without adding a 'home' button or similar.

**Comment by danni** [comment]
++ looks good

**Pull request merged**